diff -durN -x '*~' dietlibc-0.24.orig/diet.c dietlibc-0.24/diet.c
--- dietlibc-0.24.orig/diet.c	2004-12-19 14:07:34.000000000 +0100
+++ dietlibc-0.24/diet.c	2004-12-19 14:10:01.040058808 +0100
@@ -120,6 +120,7 @@
       platform[tmp2-cc+len]=0;
       if (shortplatform[0]=='i' && shortplatform[2]=='8' && shortplatform[3]=='6') shortplatform[1]='3';
       else if (!strncmp(shortplatform, "athlon", 6)) strcpy(shortplatform, "i386");
+      else if (!strncmp(shortplatform, "amd64", 5)) strcpy(shortplatform, "x86_64");
     } else {
 #ifdef __sparc__
 #ifdef __arch64__
