Index: dietlibc-0.29/lib/__nice.c
===================================================================
--- dietlibc-0.29.orig/lib/__nice.c
+++ dietlibc-0.29/lib/__nice.c
@@ -1,9 +1,14 @@
 #include "syscalls.h"
 #include <sys/time.h>
 #include <sys/resource.h>
+#include <errno.h>
 
 #ifndef __NR_nice
 int nice(int i) {
-  return setpriority(PRIO_PROCESS,0,getpriority(PRIO_PROCESS,0)+i);
+    if (setpriority(PRIO_PROCESS,0,getpriority(PRIO_PROCESS,0)+i) == -1) {
+	errno=EPERM;
+	return -1;
+    }
+    return getpriority(PRIO_PROCESS,0);
 }
 #endif
