--- dietlibc-0.33/i386/dyn_syscalls.S.orig	2014-01-28 17:41:38.966558564 +0100
+++ dietlibc-0.33/i386/dyn_syscalls.S	2014-01-28 17:58:01.516517331 +0100
@@ -195,7 +195,6 @@
 #include "../syscalls.s/n_sigpending.S"
 #include "../syscalls.s/n_sigprocmask.S"
 #include "../syscalls.s/n_sigsuspend.S"
-#include "../syscalls.s/nice.S"
 #include "../syscalls.s/pause.S"
 #include "../syscalls.s/personality.S"
 #include "../syscalls.s/query_module.S"
@@ -400,4 +401,3 @@
 #include "__longjmp.S"
 #include "setjmp.S"
 
-#include "syscalls.s/__guard.S"
--- dietlibc-0.33/x86_64/dyn_syscalls.S.orig	2012-01-24 19:24:30.000000000 +0100
+++ dietlibc-0.33/x86_64/dyn_syscalls.S	2014-01-28 17:58:08.859850356 +0100
@@ -141,7 +141,6 @@
 #include "../syscalls.s/n_sigprocmask.S"
 #include "../syscalls.s/n_sigsuspend.S"
 #include "../syscalls.s/nanosleep.S"
-#include "../syscalls.s/nice.S"
 #include "../syscalls.s/open.S"
 #include "../syscalls.s/pause.S"
 #include "../syscalls.s/personality.S"
--- dietlibc-0.33/x86_64/clock_gettime.S.orig	2012-01-26 07:19:31.000000000 +0100
+++ dietlibc-0.33/x86_64/clock_gettime.S	2014-01-28 20:34:23.670685222 +0100
@@ -11,9 +11,16 @@
 .L1:
 	push %rsi
 	push %rdi
+#ifdef __PIC__
+	movq __vdso@GOTPCREL(%rip), %rax
+	movq (%rax),%rdi
+	movq funcname@GOTPCREL(%rip), %rsi
+	call vdso_dlsym@PLT
+#else
 	movq __vdso(%rip),%rdi
 	movq $funcname, %rsi
 	call vdso_dlsym
+#endif
 	pop %rdi
 	pop %rsi
 	jmp .L2
--- dietlibc-0.33/x86_64/gettimeofday.S.orig	2012-01-24 19:24:30.000000000 +0100
+++ dietlibc-0.33/x86_64/gettimeofday.S	2014-01-28 20:35:10.746157117 +0100
@@ -11,9 +11,16 @@
 .L1:
 	push %rsi
 	push %rdi
+#ifdef __PIC__
+	movq __vdso@GOTPCREL(%rip), %rax
+	movq (%rax),%rdi
+	movq funcname@GOTPCREL(%rip), %rsi
+	call vdso_dlsym@PLT
+#else
 	movq __vdso(%rip),%rdi
 	movq $funcname, %rsi
 	call vdso_dlsym
+#endif
 	pop %rdi
 	pop %rsi
 	jmp .L2
--- dietlibc-0.33/arm/dyn_syscalls.S.orig	2012-01-24 19:24:24.000000000 +0100
+++ dietlibc-0.33/arm/dyn_syscalls.S	2014-01-28 20:58:27.842943864 +0100
@@ -104,7 +104,6 @@
 #include "../syscalls.s/n_sigprocmask.S"
 #include "../syscalls.s/n_sigsuspend.S"
 #include "../syscalls.s/nanosleep.S"
-#include "../syscalls.s/nice.S"
 #include "../syscalls.s/open.S"
 #include "../syscalls.s/pause.S"
 #include "../syscalls.s/personality.S"
