diff -urNp -x '*.orig' dietlibc-0.34.org/dietfeatures.h dietlibc-0.34/dietfeatures.h
--- dietlibc-0.34.org/dietfeatures.h	2018-02-01 03:03:38.000000000 +0100
+++ dietlibc-0.34/dietfeatures.h	2024-12-03 00:47:08.669003636 +0100
@@ -50,11 +50,6 @@
  * zero-terminate the destination string */
 /* #define WANT_FULL_POSIX_COMPAT */
 
-/* on i386, Linux has an alternate syscall method since 2002/12/16 */
-/* on my Athlon XP, it is twice as fast, but it's only in kernel 2.5 */
-/* 20040118: enabling this breaks User Mode Linux!  It's their fault. */
-#define WANT_SYSENTER
-
 #define WANT_LINKER_WARNINGS
 
 /* you need to define this if you want to run your programs with large
