--- dietlibc-0.23/diet.c.orig	2003-06-19 13:23:24.000000000 +0200
+++ dietlibc-0.23/diet.c	2003-10-05 01:35:23.068808824 +0200
@@ -117,6 +117,7 @@
       memmove(shortplatform,argv[1],(size_t)(tmp2-cc));
       platform[tmp2-cc+len]=0;
       if (shortplatform[0]=='i' && shortplatform[2]=='8' && shortplatform[3]=='6') shortplatform[1]='3';
+      else if (!strncmp(shortplatform, "athlon", 6)) strcpy(shortplatform, "i386");
     } else {
 #ifdef __sparc__
 #ifdef __arch64__
