--- dietlibc-0.29/findcflags.sh.orig	2003-05-23 15:03:57.000000000 +0200
+++ dietlibc-0.29/findcflags.sh	2003-09-13 03:46:17.000000000 +0200
@@ -1,8 +1,8 @@
 #!/bin/sh
 version=`${1:-gcc} -v 2>&1 |sed -n '/gcc version/ {s/gcc version //;p;}'`
 case $version in
-  2.9*) echo -march=i386 -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
-  3.0*) echo -march=i386 -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
-  [34]*) echo -Os -fomit-frame-pointer -falign-functions=1 -falign-jumps=1 -falign-loops=1 -mpreferred-stack-boundary=2;;
-  *) echo -O2 -pipe -fomit-frame-pointer ;;
+  2.9*) echo ${OPTFLAGS} -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
+  3.0*) echo ${OPTFLAGS} -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
+  [34]*) echo ${OPTFLAGS} -Os -fomit-frame-pointer -falign-functions=1 -falign-jumps=1 -falign-loops=1 -mpreferred-stack-boundary=2;;
+  *) echo -O2 ${OPTFLAGS} -pipe -fomit-frame-pointer ;;
 esac
--- dietlibc-0.30/Makefile.orig	2006-12-30 22:22:02.000000000 +0000
+++ dietlibc-0.30/Makefile	2006-12-30 22:29:57.000000000 +0000
@@ -100,7 +100,7 @@
 
 profiling: $(OBJDIR)/libgmon.a $(OBJDIR)/pstart.o
 
-CFLAGS=-pipe -nostdinc
+CFLAGS=-pipe -nostdinc $(OPTFLAGS)
 CROSS=
 
 CC=gcc
