Index: dietlibc-0.30/libcompat/syscall.S
===================================================================
--- dietlibc-0.30.orig/libcompat/syscall.S
+++ dietlibc-0.30/libcompat/syscall.S
@@ -130,6 +130,12 @@ syscall:
 	b    __unified_syscall
 
 #else
-		/* arch not implemented yet */
+#include <endian.h>
+	.section	.note
+#if (__WORDSIZE == 64)
+	.quad		__syscall_2_not_implemented_for_this_arch
+#else
+	.long		__syscall_2_not_implemented_for_this_arch
+#endif
 #endif
 .size	syscall, . - syscall
