--- dietlibc-0.35/Makefile.orig	2024-10-31 14:30:46.000000000 +0100
+++ dietlibc-0.35/Makefile	2025-04-10 19:58:11.446758284 +0200
@@ -118,7 +118,7 @@ all: compile_commands.json $(WHAT)
 
 profiling: $(OBJDIR)/libgmon.a $(OBJDIR)/pstart.o
 
-DEFAULTCFLAGS=-pipe -nostdinc -D_REENTRANT -ffreestanding $(EXTRACFLAGS)
+DEFAULTCFLAGS=-pipe -nostdinc -D_REENTRANT -ffreestanding $(EXTRACFLAGS) $(OPTFLAGS)
 CFLAGS=$(DEFAULTCFLAGS)
 CROSS=
 
@@ -154,14 +154,6 @@ include $(ARCH)/Makefile.add
 
 LIBMATHOBJ=$(patsubst %,$(OBJDIR)/%,$(LIBMATH))
 
-ifeq ($(CFLAGS),$(DEFAULTCFLAGS))
-CFLAGS+=-O -fomit-frame-pointer
-endif
-
-ifneq ($(DEBUG),)
-CFLAGS = -g $(EXTRACFLAGS)
-STRIP = :
-endif
 CFLAGS += -W -Wall -Wextra -Wchar-subscripts -Wmissing-prototypes -Wmissing-declarations -Wno-switch -Wno-unused -Wredundant-decls -Wshadow
 #CFLAGS += -Wsuggest-attribute=malloc
 
diff -urNp -x '*.orig' dietlibc-0.34.org/findcflags.sh dietlibc-0.34/findcflags.sh
--- dietlibc-0.34.org/findcflags.sh	2014-04-19 17:07:32.000000000 +0200
+++ dietlibc-0.34/findcflags.sh	2024-12-03 00:46:28.132368197 +0100
@@ -1,8 +1,8 @@
 #!/bin/sh
 version=`${1:-gcc} -v 2>&1 |sed -n '/gcc version/ {s/gcc version //;p;}'`
 case $version in
-  2.9*) echo -march=i386 -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
-  3.0*) echo -march=i386 -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
-  [34]*) echo -Os -fomit-frame-pointer -falign-functions=1 -falign-jumps=1 -falign-loops=1 -mpreferred-stack-boundary=4;;
-  *) echo -O2 -pipe -fomit-frame-pointer ;;
+  2.9*) echo ${OPTFLAGS} -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
+  3.0*) echo ${OPTFLAGS} -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
+  [34]*) echo ${OPTFLAGS} -Os -fomit-frame-pointer -falign-functions=1 -falign-jumps=1 -falign-loops=1 -mpreferred-stack-boundary=4;;
+  *) echo -O2 ${OPTFLAGS} -pipe -fomit-frame-pointer ;;
 esac
