--- dietlibc-0.32/include/linux/loop.h~	2003-08-19 18:58:18.000000000 +0200
+++ dietlibc-0.32/include/linux/loop.h	2011-05-30 16:50:15.748100072 +0200
@@ -30,6 +30,22 @@
   char		reserved[4];
 };
 
+struct loop_info64 {
+  __u64		lo_device;                   /* ioctl r/o */
+  __u64		lo_inode;                    /* ioctl r/o */
+  __u64		lo_rdevice;                  /* ioctl r/o */
+  __u64		lo_offset;
+  __u64		lo_sizelimit;/* bytes, 0 == max available */
+  __u32		lo_number;                   /* ioctl r/o */
+  __u32		lo_encrypt_type;
+  __u32		lo_encrypt_key_size;         /* ioctl w/o */
+  __u32		lo_flags;                    /* ioctl r/o */
+  __u8		lo_file_name[LO_NAME_SIZE];
+  __u8		lo_crypt_name[LO_NAME_SIZE];
+  __u8		lo_encrypt_key[LO_KEY_SIZE]; /* ioctl w/o */
+  __u64		lo_init[2];
+};
+
 /* Loop filter types */
 #define LO_CRYPT_NONE		0
 #define LO_CRYPT_XOR		1
@@ -40,6 +40,7 @@
 #define LO_CRYPT_IDEA		6
 #define LO_CRYPT_DUMMY		9
 #define LO_CRYPT_SKIPJACK	10
+#define LO_CRYPT_CRYPTOAPI	18
 #define MAX_LO_CRYPT		20
 
 /* IOCTL commands --- we will commandeer 0x4C ('L') */
@@ -47,6 +48,12 @@
 #define LOOP_CLR_FD	0x4C01
 #define LOOP_SET_STATUS	0x4C02
 #define LOOP_GET_STATUS	0x4C03
+#define LOOP_SET_STATUS	0x4C02
+#define LOOP_GET_STATUS	0x4C03
+#define LOOP_SET_STATUS64	0x4C04
+#define LOOP_GET_STATUS64	0x4C05
+#define LOOP_CHANGE_FD	0x4C06
+#define LOOP_SET_CAPACITY	0x4C07
 
 __END_DECLS
 
