diff -Nur dietlibc-0.32.orig/libpthread/Makefile dietlibc-0.32/libpthread/Makefile
--- dietlibc-0.32.orig/libpthread/Makefile	2003-08-18 13:32:52.000000000 -0400
+++ dietlibc-0.32/libpthread/Makefile	2010-06-21 09:42:08.000000000 -0400
@@ -2,7 +2,7 @@
 
 M=$(shell ../findcflags.sh $(CC))
 
-ARCH=$(shell uname -m | sed 's/i[4-9]86/i386/')
+ARCH=$(shell uname -m | sed -e 's/i[4-9]86/i386/' -e 's/armv[3-7]t\?[eh]\?j\?[lb]/arm/')
 
 CFLAGS= -g #-pipe -O -march=k6
 CFLAGS= -pipe
