diff -ur dietlibc-0.22/ppc/mmap.S dietlibc-0.22-/ppc/mmap.S
--- dietlibc-0.22/ppc/mmap.S	2003-07-01 13:01:26.000000000 +0200
+++ dietlibc-0.22-/ppc/mmap.S	2003-07-01 13:01:13.000000000 +0200
@@ -0,0 +1,27 @@
+        .section        ".text"
+        .align 2
+        .globl mmap
+        .type    mmap,@function
+mmap:
+        stwu 1,-16(1)
+        mflr 0
+        stw 31,12(1)
+        stw 0,20(1)
+        li 0,90
+        sc
+        mfcr 0
+        andis. 9,0,4096
+        mr 31,3
+        bc 12,2,.L4
+        bl __errno_location
+        stw 31,0(3)
+        li 31,-1
+.L4:
+        mr 3,31
+        lwz 0,20(1)
+        mtlr 0
+        lwz 31,12(1)
+        la 1,16(1)
+        blr
+.Lfe1:
+        .size    mmap,.Lfe1-mmap
--- dietlibc-0.24/include/sys/ucontext.h.orig	2003-10-10 15:17:46.000000000 +0200
+++ dietlibc-0.24/include/sys/ucontext.h	2003-11-22 18:21:22.390804792 +0100
@@ -1,6 +1,13 @@
 #ifndef _SYS_UCONTEXT_H
 #define _SYS_UCONTEXT_H
 
+/* follow glibc <bits/sigcontext.h> on ppc: */
+#if defined(powerpc) && !defined(sigcontext_struct)
+/* Kernel headers before 2.1.1 define a struct sigcontext_struct, but
+ *    we need sigcontext.  */
+# define sigcontext_struct sigcontext
+#endif
+
 #include <asm/sigcontext.h>
 #include <signal.h>
 
