diff -urNp -x '*.orig' dietlibc-0.34.org/include/fcntl.h dietlibc-0.34/include/fcntl.h
--- dietlibc-0.34.org/include/fcntl.h	2017-08-09 21:12:25.000000000 +0200
+++ dietlibc-0.34/include/fcntl.h	2024-12-03 00:46:57.932835338 +0100
@@ -55,6 +55,10 @@ __BEGIN_DECLS
 #define F_SETLK64	13
 #define F_SETLKW64	14
 
+#define F_SETLEASE	1024	/* Set a lease.  */
+#define F_GETLEASE	1025	/* Enquire what lease is active.  */
+#define F_NOTIFY	1026	/* Request notfications on a directory.  */
+
 #define FD_CLOEXEC	1	/* actually anything with low bit set goes */
 
 /* for posix fcntl() and lockf() */
@@ -81,6 +85,15 @@ __BEGIN_DECLS
 #define LOCK_WRITE	128	/* ... Which allows concurrent write operations */
 #define LOCK_RW		192	/* ... Which allows concurrent read & write ops */
 
+/* Types of directory notifications that may be requested with F_NOTIFY.  */
+#define DN_ACCESS	0x00000001	/* File accessed.  */
+#define DN_MODIFY	0x00000002	/* File modified.  */
+#define DN_CREATE	0x00000004	/* File created.  */
+#define DN_DELETE	0x00000008	/* File removed.  */
+#define DN_RENAME	0x00000010	/* File renamed.  */
+#define DN_ATTRIB	0x00000020	/* File changed attibutes.  */
+#define DN_MULTISHOT	0x80000000	/* Don't remove notifier.  */
+
 struct flock {
   int16_t l_type;
   int16_t l_whence;
@@ -137,6 +150,10 @@ struct flock64 {
 #define F_SETSIG	10	/*  for sockets. */
 #define F_GETSIG	11	/*  for sockets. */
 
+#define F_SETLEASE	1024	/* Set a lease.  */
+#define F_GETLEASE	1025	/* Enquire what lease is active.  */
+#define F_NOTIFY	1026	/* Request notfications on a directory.  */
+
 /* for F_[GET|SET]FL */
 #define FD_CLOEXEC	1	/* actually anything with low bit set goes */
 
@@ -162,6 +179,15 @@ struct flock64 {
 #define LOCK_WRITE     128     /* ... Which allows concurrent write operations */
 #define LOCK_RW        192     /* ... Which allows concurrent read & write ops */
 
+/* Types of directory notifications that may be requested with F_NOTIFY.  */
+#define DN_ACCESS	0x00000001	/* File accessed.  */
+#define DN_MODIFY	0x00000002	/* File modified.  */
+#define DN_CREATE	0x00000004	/* File created.  */
+#define DN_DELETE	0x00000008	/* File removed.  */
+#define DN_RENAME	0x00000010	/* File renamed.  */
+#define DN_ATTRIB	0x00000020	/* File changed attibutes.  */
+#define DN_MULTISHOT	0x80000000	/* Don't remove notifier.  */
+
 struct flock {
   int16_t l_type;
   int16_t l_whence;
@@ -220,6 +246,10 @@ struct flock {
 #define F_SETLKW64	35
 #endif
 
+#define F_SETLEASE	1024	/* Set a lease.  */
+#define F_GETLEASE	1025	/* Enquire what lease is active.  */
+#define F_NOTIFY	1026	/* Request notfications on a directory.  */
+
 /* for F_[GET|SET]FL */
 #define FD_CLOEXEC	1	/* actually anything with low bit set goes */
 
@@ -247,6 +277,15 @@ struct flock {
 #define LOCK_WRITE	128	/* ... Which allows concurrent write operations */
 #define LOCK_RW		192	/* ... Which allows concurrent read & write ops */
 
+/* Types of directory notifications that may be requested with F_NOTIFY.  */
+#define DN_ACCESS	0x00000001	/* File accessed.  */
+#define DN_MODIFY	0x00000002	/* File modified.  */
+#define DN_CREATE	0x00000004	/* File created.  */
+#define DN_DELETE	0x00000008	/* File removed.  */
+#define DN_RENAME	0x00000010	/* File renamed.  */
+#define DN_ATTRIB	0x00000020	/* File changed attibutes.  */
+#define DN_MULTISHOT	0x80000000	/* Don't remove notifier.  */
+
 #ifndef __mips64__
 struct flock {
   int16_t l_type;
@@ -317,6 +356,10 @@ struct flock {
 #define F_SETLK64	13
 #define F_SETLKW64	14
 
+#define F_SETLEASE	1024	/* Set a lease.  */
+#define F_GETLEASE	1025	/* Enquire what lease is active.  */
+#define F_NOTIFY	1026	/* Request notfications on a directory.  */
+
 /* for F_[GET|SET]FL */
 #define FD_CLOEXEC	1	/* actually anything with low bit set goes */
 
@@ -344,6 +387,15 @@ struct flock {
 #define LOCK_WRITE	128	/* ... Which allows concurrent write operations */
 #define LOCK_RW		192	/* ... Which allows concurrent read & write ops */
 
+/* Types of directory notifications that may be requested with F_NOTIFY.  */
+#define DN_ACCESS	0x00000001	/* File accessed.  */
+#define DN_MODIFY	0x00000002	/* File modified.  */
+#define DN_CREATE	0x00000004	/* File created.  */
+#define DN_DELETE	0x00000008	/* File removed.  */
+#define DN_RENAME	0x00000010	/* File renamed.  */
+#define DN_ATTRIB	0x00000020	/* File changed attibutes.  */
+#define DN_MULTISHOT	0x80000000	/* Don't remove notifier.  */
+
 struct flock {
   int16_t l_type;
   int16_t l_whence;
@@ -409,6 +461,10 @@ struct flock64 {
 #define F_SETLK64	13
 #define F_SETLKW64	14
 
+#define F_SETLEASE	1024	/* Set a lease.  */
+#define F_GETLEASE	1025	/* Enquire what lease is active.  */
+#define F_NOTIFY	1026	/* Request notfications on a directory.  */
+
 /* for F_[GET|SET]FL */
 #define FD_CLOEXEC	1	/* actually anything with low bit set goes */
 
@@ -436,6 +492,15 @@ struct flock64 {
 #define LOCK_WRITE	128	/* ... Which allows concurrent write operations */
 #define LOCK_RW		192	/* ... Which allows concurrent read & write ops */
 
+/* Types of directory notifications that may be requested with F_NOTIFY.  */
+#define DN_ACCESS	0x00000001	/* File accessed.  */
+#define DN_MODIFY	0x00000002	/* File modified.  */
+#define DN_CREATE	0x00000004	/* File created.  */
+#define DN_DELETE	0x00000008	/* File removed.  */
+#define DN_RENAME	0x00000010	/* File renamed.  */
+#define DN_ATTRIB	0x00000020	/* File changed attibutes.  */
+#define DN_MULTISHOT	0x80000000	/* Don't remove notifier.  */
+
 struct flock {
   int16_t l_type;
   int16_t l_whence;
@@ -505,6 +570,10 @@ struct flock64 {
 #define F_SETLKW64	14
 #endif
 
+#define F_SETLEASE	1024	/* Set a lease.  */
+#define F_GETLEASE	1025	/* Enquire what lease is active.  */
+#define F_NOTIFY	1026	/* Request notfications on a directory.  */
+
 /* for F_[GET|SET]FL */
 #define FD_CLOEXEC	1	/* actually anything with low bit set goes */
 
@@ -532,6 +601,15 @@ struct flock64 {
 #define LOCK_WRITE	128	/* ... Which allows concurrent write operations */
 #define LOCK_RW		192	/* ... Which allows concurrent read & write ops */
 
+/* Types of directory notifications that may be requested with F_NOTIFY.  */
+#define DN_ACCESS	0x00000001	/* File accessed.  */
+#define DN_MODIFY	0x00000002	/* File modified.  */
+#define DN_CREATE	0x00000004	/* File created.  */
+#define DN_DELETE	0x00000008	/* File removed.  */
+#define DN_RENAME	0x00000010	/* File renamed.  */
+#define DN_ATTRIB	0x00000020	/* File changed attibutes.  */
+#define DN_MULTISHOT	0x80000000	/* Don't remove notifier.  */
+
 struct flock {
   int16_t l_type;
   int16_t l_whence;
@@ -599,6 +677,10 @@ struct flock64 {
 #define F_SETSIG    13 /*  for sockets. */
 #define F_GETSIG    14 /*  for sockets. */
 
+#define F_SETLEASE	1024	/* Set a lease.  */
+#define F_GETLEASE	1025	/* Enquire what lease is active.  */
+#define F_NOTIFY	1026	/* Request notfications on a directory.  */
+
 #define FD_CLOEXEC  1   /* actually anything with low bit set goes */
 
 #define F_RDLCK     1   /* Read lock.  */
@@ -622,6 +704,15 @@ struct flock64 {
 #define LOCK_WRITE     128     /* ... Which allows concurrent write operations */
 #define LOCK_RW                192     /* ... Which allows concurrent read & write ops */
 
+/* Types of directory notifications that may be requested with F_NOTIFY.  */
+#define DN_ACCESS	0x00000001	/* File accessed.  */
+#define DN_MODIFY	0x00000002	/* File modified.  */
+#define DN_CREATE	0x00000004	/* File created.  */
+#define DN_DELETE	0x00000008	/* File removed.  */
+#define DN_RENAME	0x00000010	/* File renamed.  */
+#define DN_ATTRIB	0x00000020	/* File changed attibutes.  */
+#define DN_MULTISHOT	0x80000000	/* Don't remove notifier.  */
+
 
 struct flock
 {
