diff -uNrp dietlibc-0.29.orig/diet.c dietlibc-0.29/diet.c
--- dietlibc-0.32-orig/diet.c	2009-03-17 00:17:58.000000000 +0100
+++ dietlibc-0.32/diet.c	2009-08-03 13:45:59.000000000 +0200
@@ -147,6 +147,12 @@
       memmove(shortplatform,argv[1],(size_t)(tmp2-cc));
       platform[tmp2-cc+len]=0;
       if (shortplatform[0]=='i' && shortplatform[2]=='8' && shortplatform[3]=='6') shortplatform[1]='3';
+      else if (!strncmp(shortplatform, "pentium4", 8)) strcpy(shortplatform, "i386");
+      else if (!strncmp(shortplatform, "pentium3", 8)) strcpy(shortplatform, "i386");
+      else if (!strncmp(shortplatform, "athlon", 6)) strcpy(shortplatform, "i386");
+      else if (!strncmp(shortplatform, "amd64", 5)) strcpy(shortplatform, "x86_64");
+      else if (!strncmp(shortplatform, "armv", 4)) strcpy(shortplatform, "arm");
+      else if (!strncmp(shortplatform, "sparcv", 6)) strcpy(shortplatform, "sparc");
       if (!strncmp(shortplatform, "powerpc", 7)) {
         shortplatform[0] = shortplatform[1] = 'p'; shortplatform[2] = 'c';
         memmove(&shortplatform[3], &shortplatform[7], strlen(&shortplatform[7]) + 1);
