diff -ur dietlibc-0.30.org/findcflags.sh dietlibc-0.30/findcflags.sh
--- dietlibc-0.30.org/findcflags.sh	2005-04-23 13:47:18.000000000 +0200
+++ dietlibc-0.30/findcflags.sh	2007-02-06 22:24:17.449538250 +0100
@@ -1,8 +1,8 @@
 #!/bin/sh
 version=`${1:-gcc} -v 2>&1 |sed -n '/gcc version/ {s/gcc version //;p;}'`
 case $version in
-  2.9*) echo -march=i386 -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
-  3.0*) echo -march=i386 -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
-  [34]*) echo -Os -fomit-frame-pointer -falign-functions=1 -falign-jumps=1 -falign-loops=1 -mpreferred-stack-boundary=2;;
-  *) echo -O2 -pipe -fomit-frame-pointer ;;
+  2.9*) echo ${OPTFLAGS} -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
+  3.0*) echo ${OPTFLAGS} -Os -fomit-frame-pointer -malign-functions=1 -malign-jumps=1 -malign-loops=1 -mpreferred-stack-boundary=2 ;;
+  [34]*) echo ${OPTFLAGS} -Os -fomit-frame-pointer -falign-functions=1 -falign-jumps=1 -falign-loops=1 -mpreferred-stack-boundary=2;;
+  *) echo -O2 ${OPTFLAGS} -pipe -fomit-frame-pointer ;;
 esac
diff -ur dietlibc-0.30.org/Makefile dietlibc-0.30/Makefile
--- dietlibc-0.30.org/Makefile	2006-06-18 20:32:28.000000000 +0200
+++ dietlibc-0.30/Makefile	2007-02-06 22:25:12.368970500 +0100
@@ -100,7 +100,7 @@
 
 profiling: $(OBJDIR)/libgmon.a $(OBJDIR)/pstart.o
 
-CFLAGS=-pipe -nostdinc
+CFLAGS=-pipe -nostdinc $(OPTFLAGS)
 CROSS=
 
 CC=gcc
@@ -132,14 +132,6 @@
 
 LIBMATHOBJ=$(patsubst %,$(OBJDIR)/%,$(LIBMATH))
 
-ifeq ($(CFLAGS),-pipe -nostdinc)
-CFLAGS+=-O -fomit-frame-pointer
-endif
-
-ifneq ($(DEBUG),)
-CFLAGS = -g
-COMMENT = :
-endif
 CFLAGS += -W -Wall -Wextra -Wchar-subscripts -Wmissing-prototypes -Wmissing-declarations -Wno-switch -Wno-unused -Wredundant-decls
 
 PWD=$(shell pwd)
